samtranslator/swagger/swagger.py

```diff
@@ -531,7 +531,17 @@ def set_path_default_authorizer(
             if add_default_auth_to_preflight or normalized_method_name != "options":
                 normalized_method_name = self._normalize_method_name(method_name)
                 # It is possible that the method could have two definitions in a Fn::If block.
-                for method_definition in self.get_method_contents(self.get_path(path)[normalized_method_name]):
+                method_dict = self.get_path(path).get(normalized_method_name)
```

Comment:
Some additional questions which may have already been considered.

can normalized_method_name be anything other than a string?
can path be anything other than a string? 

Could we have intrinsics which resolve each of them to a dict instead of a string?